calabash-cucumber/lib/calabash-cucumber/keyboard_helpers.rb

```diff
@@ -898,6 +898,16 @@ def uia_wait_for_keyboard(opts={})
         end
       end
 
+      # Waits for a keyboard to appear and returns the localized name of the
+      # `key_code` signifier
+      #
+      # @param [String] key_code Maps to a specific name in some localization
+      def lookup_key_name(key_code)
+        wait_for_keyboard
+        localized_lang = JSON.parse(http(:path => 'keyboard-language'))['results']['input_mode']
```

Comment:
Can you pass this call through `map`?

https://github.com/calabash/calabash-ios/blob/develop/calabash-cucumber/lib/calabash-cucumber/map.rb

Alternatively, handle JSON.parse errors and the success/failure here.  The goal is to rescue errors and present helpful message to the user.  For example, if the app crashed during the http call, JSON.parse would raise a "Could not parse; expected octet blah blah blah" error.  It would be better to say something along the lines of "Could not parse output of keyboard-language route.  Did the app crash?"

Calabash 2.0 does a decent (if convoluted) job of this for iOS.

* https://github.com/calabash/calabash/blob/develop/lib/calabash/ios/device/routes/handle_route_mixin.rb#L18
* https://github.com/calabash/calabash/blob/develop/lib/calabash/ios/device/routes/response_parser.rb#L9